### ISSUE TITLE
Allow deviations to modify "when" statements

### ISSUE THREAD

**Issue by immibis** [netmod-wg/yang-next#14]
Should be self-explanatory; allow deviations to add, remove or replace `when` statements like they can with `must` statements.

**Comment by guangyingzheng** [comment]
Agree, we have the same scenario to add or modify "when" statment

**Comment by schoenw** [comment]
This may need more discussion but may be doable.

**Comment by mbj4668** [comment]
I'm not sure this a good idea. It was discussed even for YANG version 1, and since "when" changes the structure of the schema, we decided for YANG 1 that it shouldn't be allowed.  Just like changing a container to a list is also not allowed.  It seems this would change the intention of the module "too much".

**Comment by lllyfeng** [comment]
agree. Deviation can be used to delete schema, the structure of schema has already  be changed. If a schema node is deleted and it is just referenced by a when xpath expression, it will cause a error.

**Comment by abierman** [comment]
Do not support making servers support even more complex when-stmt implementation requirements
